src/encord_active/app/project_comparison/project_similarity.py

```diff
@@ -0,0 +1,165 @@
+from typing import cast
+
+import numpy as np
+import pandas as pd
+import streamlit as st
+from scipy.stats import ks_2samp
+
+from encord_active.app.common.state import get_state
+from encord_active.app.common.utils import setup_page
+from encord_active.app.projects_page import project_list
+from encord_active.lib.charts.project_similarity import (
+    ProjectSimilaritySchema,
+    plot_project_similarity_metric_wise,
+    render_2d_metric_similarity_plot,
+)
+from encord_active.lib.db.connection import DBConnection
+from encord_active.lib.db.merged_metrics import MergedMetrics
+from encord_active.lib.metrics.metadata import fetch_metrics_meta
+from encord_active.lib.project import ProjectFileStructure
+from encord_active.lib.project.metadata import fetch_project_meta
+
+# The following metrics does not give any information comparable to another project
+METRICS_TO_EXCLUDE = [
+    "Random Values on Objects",
+    "Random Values on Images",
+    "Image Difficulty",
+]
+
+
+def calculate_metric_similarity(array_1: np.ndarray, array_2: np.ndarray) -> float:
+    ks_score, _ = ks_2samp(array_1, array_2)
+    return 1 - ks_score
+
+
+def render_2d_metric_similarity_container(
+    all_metrics: list, merged_metrics_1: pd.DataFrame, merged_metrics_2: pd.DataFrame, project_name_2: str
+):
+    metrics_filtered = []
+    for metric_name in all_metrics:
+        if metric_name in merged_metrics_1 and merged_metrics_1[metric_name].dropna().shape[0] > 0:
+            metrics_filtered.append(metric_name)
+
+    if len(metrics_filtered) <= 1:
+        st.write("At least two metrics should be generated for 2D metric similarity plot")
+        return
+
+    metric_selection_col_1, metric_selection_col_2 = st.columns(2)
+    metric_name_1 = metric_selection_col_1.selectbox(
+        "Select the first metric",
+        options=metrics_filtered,
+        index=0,
+        key=f"project_comparison_metric_selection_1_{get_state().project_paths.project_dir.name}_{project_name_2}",
+    )
+
+    # Second selectbox should be populated according to the selected metric in the first one
+    # due to different metric types. E.g., 'Area' is not compatible with 'object aspect ratio'
+    options_for_second_metric = []
+
+    for metric_name in metrics_filtered:
+        if metric_name == metric_name_1:
+            continue
+        metric_df_1 = merged_metrics_1[metric_name_1].dropna().to_frame()
+        metric_df_2 = merged_metrics_1[metric_name].dropna().to_frame()
+
+        merged_metric = metric_df_1.join(metric_df_2, how="inner")
+        if merged_metric.shape[0] > 0:
+            options_for_second_metric.append(metric_name)
```

Comment:
Even after fixing the typo below, I don't understand the purpose of this piece of code. For two different projects, the merging of the identifiers should always have shape `[0, 2]` because they don't share label hashes. So the result will always be an empty list.

```suggestion
    metric_df_1 = merged_metrics_1[metric_name_1].dropna().to_frame()
    for metric_name in metrics_filtered:
        if metric_name == metric_name_1:
            continue
        metric_df_2 = merged_metrics_2[metric_name].dropna().to_frame()

        merged_metric = metric_df_1.join(metric_df_2, how="inner")
        if merged_metric.shape[0] > 0:
            options_for_second_metric.append(metric_name)
```

Reply:
Actually, it is not a typo; it was intentional. The aim here is to find compatible metrics with each other that can be drawn on 2D plot. It iterates through metrics and sees if its identifier is compatible to the chosen one.

Reply:
See line 71